### PR TITLE
integrating fluxible-router

### PR DESCRIPTION
**Pull request by jedireza**
Still to do:

 - [x] fix styling issues
    - For some reason the nav links are not floating right and the logo text isn't always hiding when it should.

~~Depends on update to `fluxible-router`: https://github.com/yahoo/fluxible-router/pull/12~~

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by redonkulus** [comment]
:shipit:

**Comment by jedireza** [comment]
Because we're moving away from https://github.com/yahoo/fluxible-router/pull/12 in favor of a less opinionated router, I'll be updating this PR to follow the ideas in https://github.com/yahoo/fluxible-router/pull/14.

**Comment by jedireza** [comment]
Depends on yahoo/fluxible-router#14 before we can merge.

**Comment by mridgway** [comment]
:+1:

File: components/Application.jsx [hunk 1]
```diff
@@ -69,38 +65,33 @@ var Application = React.createClass({
...
+                            </NavLink> <TopNav selected={routeName} />
                         </div>
                     </div>
-                    <Component doc={this.state.currentDoc} currentRoute={this.state.route} />
+                    {Handler}
                 </div>
                 <div id="footer" className="P-20px Bdt-1" role="footer">
                     <div className="innerwrapper spaceBetween Mx-a--sm W-90%--sm W-a--sm">
-                        <small>All code on this site is licensed under the <a href="https://github.com/yahoo/fluxible.io/blob/master/LICENSE.md">Yahoo BSD License</a>, unless otherwise stated.</small> <small>&copy; 2015 Yahoo! Inc. All rights reserved.</small>
+                        <small>All code on this site is licensed under the <a href="https://github.com/yahoo/fluxible.io/blob/master/LICENSE.md">Yahoo BSD License</a>, unless otherwise stated. <NavLink href="/demo-err-404.html">Demo 404</NavLink> | <NavLink href="/demo-err-500.html">Demo 500</NavLink></small> <small>&copy; 2015 Yahoo! Inc. All rights reserved.</small>
```
Comment (redonkulus) [inline components/Application.jsx]: Can we move these off the footer and put into reference apps or some other page. Having them on the footer of ever page looks like a bug.

File: package.json [hunk 2]
```diff
@@ -40,44 +40,42 @@
     }
   },
   "dependencies": {
-    "babel": "^4.7.8",
+    "babel": "^5.1.11",
```
Comment (redonkulus) [inline package.json]: we have to test this internally, as this will break.

File: components/Application.jsx (hunk 1 above)
Comment (jedireza) [inline components/Application.jsx]: I'm open to suggestions on where to put these links. Keeping them is good for demonstrating their functionality.

File: package.json (hunk 2 above)
Comment (jedireza) [inline package.json]: Roger that. I plan to test the workaround we discussed earlier today.

File: components/Application.jsx (hunk 1 above)
Comment (redonkulus) [inline components/Application.jsx]: Put them on the Reference Apps page for now, just not in the footer.

**Comment by redonkulus** [comment]
:+1: once the demo links are moved.

**Comment by redonkulus** [comment]
:+1:

**Pull request merged**